src/gui/main.cpp

```diff
@@ -297,60 +297,40 @@ int main(int argc, char **argv)
-                // If the systemtray is not there, we will wait one second for it to maybe start
-                // (eg boot time) then we show the settings dialog if there is still no systemtray.
-                // On XFCE however, we show a message box with explainaition how to install a systemtray.
-                qCInfo(lcMain) << "System tray is not available, waiting...";
-                Utility::sleep(1);
-
-                auto desktopSession = qgetenv("XDG_CURRENT_DESKTOP").toLower();
-                if (desktopSession.isEmpty()) {
-                    desktopSession = qgetenv("DESKTOP_SESSION").toLower();
-                }
-                if (desktopSession == "xfce") {
-                    int attempts = 0;
-                    while (!QSystemTrayIcon::isSystemTrayAvailable()) {
-                        attempts++;
-                        if (attempts >= 30) {
-                            qCWarning(lcMain) << "System tray unavailable (xfce)";
-                            warnSystray();
-                            break;
-                        }
-                        Utility::sleep(1);
-                    }
-                }
 
-                if (QSystemTrayIcon::isSystemTrayAvailable()) {
-                    ocApp->tryTrayAgain();
-                } else if (desktopSession != "ubuntu") {
-                    qCInfo(lcMain) << "System tray still not available, showing window and trying again later";
-                    QTimer::singleShot(10s, ocApp, &Application::tryTrayAgain);
-                } else {
-                    qCInfo(lcMain) << "System tray still not available, but assuming it's fine on 'ubuntu' desktop";
+        if (!QSystemTrayIcon::isSystemTrayAvailable()) {
+            // If the systemtray is not there, we will wait one second for it to maybe start
+            // (eg boot time) then we show the settings dialog if there is still no systemtray.
+            // On XFCE however, we show a message box with explainaition how to install a systemtray.
+            qCInfo(lcMain) << "System tray is not available, waiting...";
+            Utility::sleep(1);
+
+            auto desktopSession = qgetenv("XDG_CURRENT_DESKTOP").toLower();
+            if (desktopSession.isEmpty()) {
+                desktopSession = qgetenv("DESKTOP_SESSION").toLower();
+            }
+            if (desktopSession == "xfce") {
+                int attempts = 0;
+                while (!QSystemTrayIcon::isSystemTrayAvailable()) {
+                    attempts++;
+                    if (attempts >= 30) {
+                        qCWarning(lcMain) << "System tray unavailable (xfce)";
+                        warnSystray();
+                        break;
+                    }
+                    Utility::sleep(1);
                 }
             }
-        }
-    } else {
-        // if the application is already running, notify it.
-        qCInfo(lcMain) << "Already running, exiting...";
```

Comment:
Why was the single instance ipc removed?

Reply:
done